include/Dialect/Btor/IR/BtorOps.td

```diff
@@ -1048,9 +1054,10 @@ def WriteOp : Btor_Op<"write"> {
     ```
   }];
 
-  let arguments = (ins SignlessIntegerLike:$value, AnyVector:$base,
+  let arguments = (ins SignlessIntegerLike:$value, Btor_Array:$base,
                        SignlessIntegerLike:$index);
 
+  //will break due to custom btor array type
```

Comment:
The reason why this breaks could do with the fact that you force a type cast in the `getArrayType` function.